File: strands-py/src/strands/agent/conversation_manager/null_conversation_manager.py
"""Null implementation of conversation management."""

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from ...agent.agent import Agent

from .conversation_manager import ConversationManager


class NullConversationManager(ConversationManager):
    """A no-op conversation manager that does not modify the conversation history.

    Useful for:

    - Testing scenarios where conversation management should be disabled
    - Cases where conversation history is managed externally
    - Situations where the full conversation history should be preserved
    """

    def apply_management(self, agent: "Agent", **kwargs: Any) -> None:
        """Does nothing to the conversation history.

        Args:
            agent: The agent whose conversation history will remain unmodified.
            **kwargs: Additional keyword arguments for future extensibility.
        """
        pass

    def reduce_context(self, agent: "Agent", e: Exception | None = None, **kwargs: Any) -> None:
        """Does not reduce context.

        When called reactively (e is not None), re-raises the overflow exception since this
        manager cannot reduce context. When called proactively (e is None), returns silently.

        Args:
            agent: The agent whose conversation history will remain unmodified.
            e: The exception that triggered the context reduction, if any.
            **kwargs: Additional keyword arguments for future extensibility.

        Raises:
            e: If provided (reactive overflow).
        """
        if e:
            raise e

File: strands-ts/src/conversation-manager/null-conversation-manager.ts
/**
 * Null implementation of conversation management.
 *
 * This module provides a no-op conversation manager that does not modify
 * the conversation history. Useful for testing and scenarios where conversation
 * management is handled externally.
 */

import { ConversationManager, type ConversationManagerReduceOptions } from './conversation-manager.js'

/**
 * A no-op conversation manager that does not modify the conversation history.
 *
 * Does not register any proactive hooks. Overflow errors will not be retried
 * since `reduce` always returns `false`.
 */
export class NullConversationManager extends ConversationManager {
  /**
   * Unique identifier for this conversation manager.
   */
  readonly name = 'strands:null-conversation-manager'

  /**
   * No-op reduction — never modifies the conversation history.
   *
   * @returns `false` always
   */
  reduce(_args: ConversationManagerReduceOptions): boolean {
    return false
  }
}
